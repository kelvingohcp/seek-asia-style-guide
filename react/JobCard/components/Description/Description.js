import React from 'react';
import PropTypes from 'prop-types';
import styles from './Description.less';
import classnames from 'classnames';
import Text from '../../../Text/Text';

const Description = ({ description, showDescription }) => {
  if (!(showDescription && description)) {
    return null;
  }

  return (
    <div className={classnames(styles.desktopOnly, styles.description)}>
      <Text
        whispering
        baseline={false}
        className={styles.text}>
        {description}
      </Text>
    </div>
  );
};

Description.propTypes = {
  description: PropTypes.string,
  showDescription: PropTypes.bool
};

export default Description;
