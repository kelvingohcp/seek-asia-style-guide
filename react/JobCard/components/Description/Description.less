@import (reference) '~seek-asia-style-guide/theme';
@import (reference) '../../JobCard';

.text {
  color: @saGrey1;
  text-overflow: ellipsis;
  overflow: hidden;
  padding-bottom: @grid-base;
}

.desktopOnly {
  display: none;
  @media @aboveMobile {
    display: flex;
  }
}

.description {
  padding-top: @grid-base * 4;
  @media @aboveMobile {
    padding-top: @grid-base * 2;
    margin-right: @grid-base * 4;
  }
}
