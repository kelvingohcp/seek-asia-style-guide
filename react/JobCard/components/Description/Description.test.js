import React from 'react';
import { shallow } from 'enzyme';
import Description from './Description';

const description = 'dummyDescription';

describe('Description', () => {
  it('should render correctly', () => {
    expect(shallow(<Description description={description} showDescription={true} />)).toMatchSnapshot();
  });

  it('should not show description when showDescription is false', () => {
    expect(shallow(<Description description={description} showDescription={false} />)).toMatchSnapshot();
  });
});
