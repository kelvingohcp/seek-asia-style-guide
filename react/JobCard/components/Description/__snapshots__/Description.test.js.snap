// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should not show description when showDescription is false 1`] = `""`;

exports[`Description should render correctly 1`] = `
<div
  className="desktopOnly description"
>
  <Text
    baseline={false}
    className="text"
    whispering={true}
  >
    dummyDescription
  </Text>
</div>
`;
